packages/babel-plugin-proposal-object-rest-spread/src/index.ts

```diff
@@ -385,12 +413,17 @@ export default declare((api, opts) => {
             );
           });
 
-          const objectPatternPath = path.findParent(path =>
-            path.isObjectPattern(),
+          //@ts-expect-error: findParent can not apply assertions on result shape
+          const objectPatternPath: NodePath<t.ObjectPattern> = path.findParent(
```

Comment:
Not in this PR, but maybe something like this would work:
```js
function findParent<T extends t.Node>(test: (path: NodePath<t.Node>) => path is NodePath<T>): T | null;
```